torch/_dynamo/variables/user_defined.py

```diff
@@ -328,15 +330,18 @@ def call_function(
             and hasattr(
                 self.value, "__exit__"
             )  # TODO(voz): These can invoke user code!
-            and check_constant_args(args, kwargs)
-            and self.value.__init__ == object.__init__
-            and len(kwargs) == 0  # TODO(ybliang): support kwargs
+            and self.value.__new__ == object.__new__
+            and SideEffects.cls_supports_mutation_side_effects(self.value)
+            and self.source is not None
```

Comment:
@zou3519, with these changes, some contexts managers from pytest/unittest are now reaching this elif statement, rather than becoming a graph break. The problematic ones are failing because they check if the test raises a specific exception (`assert_raises`, `assertRaisesRegexHighlighted`, etc). With that said, I had to add more than 100 files to `dynamo_expected_failure` folder. Would it be ok if I just ignore such context managers in this elif statement?

```python
elif (
    issubclass(type(self.value), type)
    and hasattr(
        self.value, "__enter__"
    )  # TODO(voz): These can invoke user code!
    and hasattr(
        self.value, "__exit__"
    )  # TODO(voz): These can invoke user code!
    and self.value.__new__ == object.__new__
    and SideEffects.cls_supports_mutation_side_effects(self.value)
    and self.source is not None
    and self.value not in (assert_raises, assertRaisesRegexHighlight)  # <----- this
):

```

Reply:
My motivation for this change is as while dynamo cannot handle those context managers, the test might be composed of other compilable code that has value. Adding the test to `dynamo_expected_failure` may hide bugs that we don't catch because the test will be ignored.

Reply:
@yanboliang is there a way to mark a single API from e.g. pytest as skipped?

Reply:
NB: I'm fine with this, we just need to skip 3 decorators here